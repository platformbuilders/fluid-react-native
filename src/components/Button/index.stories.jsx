import React from 'react';
import { storiesOf } from '@storybook/react-native';
import { withKnobs, text, boolean, select } from '@storybook/addon-knobs';
import { action } from '@storybook/addon-actions';
import Button from './index';
import buttonMarkdown from './button.md';
import { colors } from '../../theme';

const optionsVariant = {
  primary: colors.primary.main,
  secondary: colors.secondary.main,
  tertiary: colors.tertiary.main,
  accent: colors.accent.main,
  flat: 'transparent',
};

storiesOf('Button', module)
  .addDecorator(withKnobs)
  .addParameters({
    readme: {
      content: buttonMarkdown,
      sidebar: buttonMarkdown,
    },
    viewport: {
      defaultViewport: '',
    },
  })
  .add('Simple example', () => (
    <Button
      tertiary={boolean('tertiary', false)}
      secondary={boolean('secondary', false)}
      style={{
        width: 100,
        height: 50,
        backgroundColor: select('variant', optionsVariant, 'primary'),
      }}
      disabled={boolean('disabled', false)}
      rounded={boolean('rounded', false)}
      onPress={action('button-click')}
    >
      {text('Text', 'Button Test')}
    </Button>
  ));
