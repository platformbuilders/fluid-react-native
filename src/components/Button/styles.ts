import styled from 'styled-components/native';
import {
  ButtonColorType,
  ButtonVariants,
  ThemeProps,
  TypographyVariants,
  getTheme,
  ifStyle,
} from '@platformbuilders/theme-toolkit';

import DefaultIcon from '../Icon';
import LoadingIndicator from '../LoadingIndicator';
import TouchableComponent from '../Touchable';
import TypographyComponent from '../Typography';

const borderWidthSmall = getTheme('borderWidth.sm');
const brandPrimary = getTheme('brand.primary.main');
const brandPrimaryContrast = getTheme('brand.primary.contrast');
const brandSecondary = getTheme('brand.secondary.main');
const brandSecondaryContrast = getTheme('brand.secondary.contrast');
const brandAccent = getTheme('brand.accent.main');
const brandAccentContrast = getTheme('brand.accent.contrast');
const infoMain = getTheme('info.main');
const infoContrast = getTheme('info.contrast');
const dangerMain = getTheme('danger.main');
const dangerContrast = getTheme('danger.contrast');
const successMain = getTheme('success.main');
const successContrast = getTheme('success.contrast');
const warningMain = getTheme('warning.main');
const warningContrast = getTheme('warning.contrast');
const buttonRadius = getTheme('borderRadius.sm');
const minimumSpacing = getTheme('spacing.xs');
const smallSpacing = getTheme('spacing.sm');
const isLeftIcon = ifStyle('leftIcon');
const isRightIcon = ifStyle('rightIcon');
const hasBorder = ifStyle('hasBorder');
const isFlat = ifStyle('flat');
const isDisabled = ifStyle('disabled');

type ButtonWrapperProps = {
  rounded: boolean;
  hasBorder: boolean;
  flat: boolean;
  buttonVariant: ButtonVariants;
  colorVariant: ButtonColorType;
  disabled?: boolean;
  style: any;
  minWidth?: string | number;
  maxWidth?: string | number;
} & ThemeProps;

const buttonSize = 45;

function isTintVariant(bandSelected, props) {
  if (props.buttonVariant === 'tint') {
    return `${bandSelected(props)}10`;
  } else {
    return `${bandSelected(props)}`;
  }
}

const getBackgroundColor = (props: ButtonWrapperProps): string => {
  if (
    props.buttonVariant === 'flat' ||
    props.buttonVariant === 'ghost' ||
    props.buttonVariant === 'outline'
  ) {
    return 'transparent';
  }

  switch (props.colorVariant) {
    case 'primary':
      return isTintVariant(brandPrimary, props);
    case 'secondary':
      return isTintVariant(brandSecondary, props);
    case 'accent':
      return isTintVariant(brandAccent, props);
    case 'info':
      return isTintVariant(infoMain, props);
    case 'warning':
      return isTintVariant(warningMain, props);
    case 'danger':
      return isTintVariant(dangerMain, props);
    case 'success':
      return isTintVariant(successMain, props);
    default:
      return `${brandPrimary(props)}`;
  }
};

type TextButtonProps = {
  buttonVariant: ButtonVariants;
  colorVariant: ButtonColorType;
  variant?: TypographyVariants;
  disabled?: boolean;
  flat?: boolean;
  style: any;
} & ThemeProps;

const getTextColor = (props: TextButtonProps): string => {
  switch (props.colorVariant) {
    case 'primary':
      return `${
        props.buttonVariant === 'filled'
          ? brandPrimaryContrast(props)
          : brandPrimary(props)
      }`;
    case 'secondary':
      return `${
        props.buttonVariant === 'filled'
          ? brandSecondaryContrast(props)
          : brandSecondary(props)
      }`;
    case 'accent':
      return `${
        props.buttonVariant === 'filled'
          ? brandAccentContrast(props)
          : brandAccent(props)
      }`;
    case 'danger':
      return `${
        props.buttonVariant === 'filled'
          ? dangerContrast(props)
          : dangerMain(props)
      }`;
    case 'info':
      return `${
        props.buttonVariant === 'filled' ? infoContrast(props) : infoMain(props)
      }`;
    case 'warning':
      return `${
        props.buttonVariant === 'filled'
          ? warningContrast(props)
          : warningMain(props)
      }`;
    case 'success':
      return `${
        props.buttonVariant === 'filled'
          ? successContrast(props)
          : successMain(props)
      }`;
    default:
      return `${brandPrimaryContrast(props)}`;
  }
};

type TouchableProps = {
  rounded: boolean;
} & ThemeProps;

export const Touchable = styled(TouchableComponent)<TouchableProps>`
  border-radius: ${(props: TouchableProps): any =>
    props.rounded ? buttonSize / 2 : buttonRadius(props)}px;
`;

export const ButtonWrapper = styled.View<ButtonWrapperProps>`
  height: ${buttonSize}px;
  flex-direction: row;
  align-items: center;
  min-width: ${({ minWidth }: ButtonWrapperProps): string =>
    `${minWidth || '180px'}`};
  max-width: ${({ maxWidth }: ButtonWrapperProps) => maxWidth || '100%'};
  overflow: hidden;
  padding: ${(props: ButtonWrapperProps): string =>
    props.rounded ? '0' : `${minimumSpacing(props)}px`};
  border-radius: ${(props: ButtonWrapperProps): any =>
    props.rounded ? buttonSize / 2 : buttonRadius(props)}px;
  justify-content: center;
  background-color: ${isFlat('transparent', getBackgroundColor)};
  border-color: ${getTextColor};
  border-width: ${hasBorder(borderWidthSmall, '0')}px;
  opacity: ${isDisabled(0.3, 1)};
`;

export const TextButton = styled(TypographyComponent)<TextButtonProps>`
  letter-spacing: 0.4px;
  color: ${isFlat(getBackgroundColor, getTextColor)};
  font-weight: bold;
`;

export const Loading = styled(LoadingIndicator).attrs({
  variant: 'button',
})`
  align-self: center;
  width: 55px;
`;

type IconProps = {
  rightIcon?: boolean;
  leftIcon?: boolean;
  colorVariant: ButtonColorType;
  buttonVariant: ButtonVariants;
  style: any;
} & ThemeProps;

export const Icon = styled(DefaultIcon).attrs((props: IconProps) => ({
  color: getTextColor(props),
}))<IconProps>`
  margin-right: ${isLeftIcon(smallSpacing, 0)}px;
  margin-left: ${isRightIcon(smallSpacing, 0)}px;
`;
