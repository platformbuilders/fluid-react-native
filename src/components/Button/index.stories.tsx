import type { Meta } from '@storybook/react';
import { VARIANT_SIZE } from '../../enums/size.enum';
import Button from '.';

const meta: Meta<typeof Button> = {
  title: 'Button',
  component: Button,
  argTypes: {
    onPress: { action: 'clicked' },
    typographyVariant: { control: { type: 'select' }, options: VARIANT_SIZE },
    variant: {
      control: { type: 'select' },
      options: ['filled', 'ghost', 'tint', 'outline', 'flat'],
    },
    colorVariant: {
      control: { type: 'select' },
      options: [
        'primary',
        'secondary',
        'accent',
        'danger',
        'success',
        'warning',
        'info',
      ],
    },
  },
  args: {
    variant: 'filled',
    colorVariant: 'primary',
    accessibility: 'Button',
    accessibilityLabel: 'Button',
    disabled: false,
    onPress: () => {},
    testID: 'button-test-id',
    children: 'Ver Usuário',
    flat: false,
    id: 'button-id',
    loading: false,
    maxWidth: '250px',
    minWidth: '250px',
    rounded: false,
    leftIconName: '',
    rightIconName: '',
    typographyVariant: 'lg',
  },
  parameters: {
    notes:
      'Esse componente tem como função auxiliar na manipulação, manutenção e padronização de botões dentro de uma aplicação.',
  },
};

export default meta;

export const Default = {};

export const IconRight = {
  args: {
    ...meta.args,
    children: 'Adicionar Usuário',
    rightIconName: 'user-plus',
  },
};

export const IconLeft = {
  args: {
    ...meta.args,
    children: 'Remover Usuário',
    leftIconName: 'user-times',
  },
};

export const AccessibleButton = {
  args: {
    ...meta.args,
    children: 'Botão Acessível',
    accessibilityLabel: 'Botão com recursos completos de acessibilidade',
    accessibilityHint: 'Ativa a funcionalidade principal',
    variant: 'filled',
    colorVariant: 'primary',
  },
  parameters: {
    notes: `
      ## Acessibilidade
      
      Este botão implementa as seguintes características de acessibilidade:
      
      - **accessibilityLabel**: Descrição que será lida por leitores de tela
      - **accessibilityRole**: Define o papel como "button" para leitores de tela
      - **accessibilityState**: Informa estados como disabled e busy (quando loading)
      - **accessibilityHint**: Explica o que acontecerá ao ativar o botão
      - **hitSlop**: Área de toque aumentada para facilitar o uso
      
      ### Exemplo de uso:
      
      \`\`\`jsx
      <Button
        accessibilityLabel="Finalizar compra"
        accessibilityHint="Conclui sua compra e vai para a página de pagamento"
        onPress={handleCheckout}
      >
        Finalizar Compra
      </Button>
      \`\`\`
    `,
  },
};

export const DisabledWithAccessibility = {
  args: {
    ...meta.args,
    children: 'Botão Desabilitado',
    accessibilityLabel: 'Botão desabilitado até preencher o formulário',
    disabled: true,
  },
  parameters: {
    notes: `
      É importante fornecer informações claras para usuários com deficiência sobre por que um botão 
      está desabilitado e o que é necessário fazer para habilitá-lo.
    `,
  },
};

export const LoadingWithAccessibility = {
  args: {
    ...meta.args,
    children: 'Carregando...',
    loading: true,
    accessibilityState: { busy: true },
    accessibilityLabel: 'Botão em estado de carregamento',
  },
  parameters: {
    notes: `
      O estado de loading é informado aos leitores de tela através da propriedade
      accessibilityState={{ busy: true }}, o que ajuda usuários com deficiência visual a 
      entender que uma ação está em andamento.
    `,
  },
};
