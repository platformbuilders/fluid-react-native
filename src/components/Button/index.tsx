import React, { FC } from 'react';
import { ButtonProps } from '../../types';
import { Touchable, ButtonWrapper, TextButton, Loading } from './styles';

const Button: FC<ButtonProps> = ({
  id,
  children,
  onPress,
  accessibility,
  accessibilityLabel,
  testID,
  style = [{}],
  textStyle = {},
  disabled = false,
  rounded = false,
  loading = false,
  contrast = false,
  variant = 'primary',
  typographyVariant = 'body',
}) => {
  return (
    <Touchable
      id={id || accessibility}
      accessibility={accessibility}
      accessibilityLabel={accessibilityLabel || accessibility}
      testID={testID || id || accessibility}
      disabled={loading || disabled}
      onPress={onPress}
      rounded={rounded}
    >
      <ButtonWrapper
        buttonVariant={variant}
        style={style}
        disabled={disabled}
        rounded={rounded}
      >
        {loading && <Loading contrast={contrast} />}
        {!loading && (
          <>
            <TextButton
              style={textStyle}
              disabled={disabled}
              variant={typographyVariant}
              buttonVariant={variant}
            >
              {children}
            </TextButton>
          </>
        )}
      </ButtonWrapper>
    </Touchable>
  );
};

export default Button;
