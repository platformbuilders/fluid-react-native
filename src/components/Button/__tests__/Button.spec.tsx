import 'jest';
import React from 'react';
import renderer from 'react-test-renderer';
import { render, fireEvent } from 'react-native-testing-library';
import { ThemeProvider } from 'styled-components/native';
import Button from '..';
import { theme } from '../../../test/helpers';

describe('<Button />', () => {
  it('should render button', () => {
    const wrapper = renderer.create(
      <ThemeProvider theme={theme}>
        <Button id="testing" accessibility="" />
      </ThemeProvider>,
    );

    expect(wrapper.toJSON()).toMatchSnapshot();
  });

  it('should render button when rounded', () => {
    const wrapper = renderer.create(
      <ThemeProvider theme={theme}>
        <Button id="testing" accessibility="" rounded />
      </ThemeProvider>,
    );

    expect(wrapper.toJSON()).toMatchSnapshot();
  });

  it('should render button with loading', () => {
    const wrapper = renderer.create(
      <ThemeProvider theme={theme}>
        <Button id="testing" accessibility="" loading />
      </ThemeProvider>,
    );

    expect(wrapper.toJSON()).toMatchSnapshot();
  });

  it('should fire onPress from Button', () => {
    const onPressEvent = jest.fn();

    const { getByTestId } = render(
      <ThemeProvider theme={theme}>
        <Button id="testing" accessibility="" onPress={onPressEvent} />
      </ThemeProvider>,
    );

    const component = getByTestId('testing');

    expect(component).not.toBe(null);

    fireEvent.press(component);

    expect(onPressEvent).toHaveBeenCalled();
  });
});
