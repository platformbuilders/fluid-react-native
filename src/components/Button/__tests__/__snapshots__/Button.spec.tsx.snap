// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should render button 1`] = `
<View
  accessibilityLabel=""
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "borderRadius": 0,
      "opacity": 1,
    }
  }
  testID="testing"
>
  <View
    disabled={false}
    rounded={false}
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#C6B09E",
          "borderColor": "black",
          "borderRadius": 11,
          "borderStyle": "solid",
          "borderWidth": 0,
          "flexDirection": "row",
          "height": 75.42857142857143,
          "justifyContent": "center",
          "marginVertical": 9.428571428571429,
          "minWidth": 282.8571428571429,
          "paddingBottom": 10,
          "paddingLeft": 11,
          "paddingRight": 11,
          "paddingTop": 10,
        },
        Object {},
      ]
    }
    variant="primary"
  >
    <Text
      disabled={false}
      style={
        Array [
          Object {
            "color": "#212121",
            "fontSize": 31,
            "lineHeight": 43,
          },
          Object {
            "color": "#ffffff",
            "fontWeight": "500",
            "letterSpacing": 0.4,
          },
          Object {},
        ]
      }
      variant="body"
    />
  </View>
</View>
`;

exports[`<Button /> should render button when rounded 1`] = `
<View
  accessibilityLabel=""
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "borderRadius": 50,
      "opacity": 1,
    }
  }
  testID="testing"
>
  <View
    disabled={false}
    rounded={true}
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#C6B09E",
          "borderColor": "black",
          "borderRadius": 37.714285714285715,
          "borderStyle": "solid",
          "borderWidth": 0,
          "flexDirection": "row",
          "height": 75.42857142857143,
          "justifyContent": "center",
          "marginVertical": 9.428571428571429,
          "minWidth": 282.8571428571429,
          "paddingBottom": 0,
          "paddingLeft": 0,
          "paddingRight": 0,
          "paddingTop": 0,
        },
        Object {},
      ]
    }
    variant="primary"
  >
    <Text
      disabled={false}
      style={
        Array [
          Object {
            "color": "#212121",
            "fontSize": 31,
            "lineHeight": 43,
          },
          Object {
            "color": "#ffffff",
            "fontWeight": "500",
            "letterSpacing": 0.4,
          },
          Object {},
        ]
      }
      variant="body"
    />
  </View>
</View>
`;

exports[`<Button /> should render button with loading 1`] = `
<View
  accessibilityLabel=""
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "borderRadius": 0,
      "opacity": 1,
    }
  }
  testID="testing"
>
  <View
    disabled={false}
    rounded={false}
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#C6B09E",
          "borderColor": "black",
          "borderRadius": 11,
          "borderStyle": "solid",
          "borderWidth": 0,
          "flexDirection": "row",
          "height": 75.42857142857143,
          "justifyContent": "center",
          "marginVertical": 9.428571428571429,
          "minWidth": 282.8571428571429,
          "paddingBottom": 10,
          "paddingLeft": 11,
          "paddingRight": 11,
          "paddingTop": 10,
        },
        Object {},
      ]
    }
    variant="primary"
  >
    <View
      style={
        Array [
          Object {
            "aspectRatio": 1,
          },
          undefined,
          Array [
            Object {},
            Object {
              "alignSelf": "center",
              "width": 86.42857142857142,
            },
          ],
        ]
      }
    >
      <View
        accessibilityLabel="Aguarde"
        autoPlay={true}
        contrast={false}
        enableMergePathsAndroidForKitKatAndAbove={false}
        forwardedRef={[Function]}
        loop={true}
        onAnimationFinish={[Function]}
        onLayout={[Function]}
        progress={0}
        resizeMode="contain"
        sourceJson="{\\"v\\":\\"4.6.0\\",\\"fr\\":29.9700012207031,\\"ip\\":0,\\"op\\":49.0000019958109,\\"w\\":200,\\"h\\":200,\\"nm\\":\\"loading_ring_medium\\",\\"ddd\\":0,\\"assets\\":[],\\"layers\\":[{\\"ddd\\":0,\\"ind\\":1,\\"ty\\":4,\\"nm\\":\\"green ring 1\\",\\"ks\\":{\\"o\\":{\\"a\\":0,\\"k\\":100},\\"r\\":{\\"a\\":1,\\"k\\":[{\\"i\\":{\\"x\\":[0.833],\\"y\\":[0.833]},\\"o\\":{\\"x\\":[0.167],\\"y\\":[0.167]},\\"n\\":[\\"0p833_0p833_0p167_0p167\\"],\\"t\\":0,\\"s\\":[0],\\"e\\":[360]},{\\"t\\":49.0000019958109}]},\\"p\\":{\\"a\\":0,\\"k\\":[100,100,0]},\\"a\\":{\\"a\\":0,\\"k\\":[0,0,0]},\\"s\\":{\\"a\\":0,\\"k\\":[200,200,100]}},\\"ao\\":0,\\"shapes\\":[{\\"ty\\":\\"gr\\",\\"it\\":[{\\"d\\":1,\\"ty\\":\\"el\\",\\"s\\":{\\"a\\":0,\\"k\\":[54,54]},\\"p\\":{\\"a\\":0,\\"k\\":[0,0]},\\"nm\\":\\"Ellipse Path 1\\",\\"mn\\":\\"ADBE Vector Shape - Ellipse\\"},{\\"ty\\":\\"st\\",\\"c\\":{\\"a\\":0,\\"k\\":[0.949,0.6,0.29,1]},\\"o\\":{\\"a\\":0,\\"k\\":100},\\"w\\":{\\"a\\":0,\\"k\\":6},\\"lc\\":2,\\"lj\\":1,\\"ml\\":4,\\"nm\\":\\"Stroke 1\\",\\"mn\\":\\"ADBE Vector Graphic - Stroke\\"},{\\"ty\\":\\"tr\\",\\"p\\":{\\"a\\":0,\\"k\\":[0,0],\\"ix\\":2},\\"a\\":{\\"a\\":0,\\"k\\":[0,0],\\"ix\\":1},\\"s\\":{\\"a\\":0,\\"k\\":[100,100],\\"ix\\":3},\\"r\\":{\\"a\\":0,\\"k\\":0,\\"ix\\":6},\\"o\\":{\\"a\\":0,\\"k\\":100,\\"ix\\":7},\\"sk\\":{\\"a\\":0,\\"k\\":0,\\"ix\\":4},\\"sa\\":{\\"a\\":0,\\"k\\":0,\\"ix\\":5},\\"nm\\":\\"Transform\\"}],\\"nm\\":\\"Ellipse 1\\",\\"np\\":3,\\"mn\\":\\"ADBE Vector Group\\"},{\\"ty\\":\\"tm\\",\\"s\\":{\\"a\\":1,\\"k\\":[{\\"i\\":{\\"x\\":[0.667],\\"y\\":[1]},\\"o\\":{\\"x\\":[0.644],\\"y\\":[0]},\\"n\\":[\\"0p667_1_0p644_0\\"],\\"t\\":10,\\"s\\":[0],\\"e\\":[100]},{\\"t\\":50.0000020365418}],\\"ix\\":1},\\"e\\":{\\"a\\":1,\\"k\\":[{\\"i\\":{\\"x\\":[0.667],\\"y\\":[1]},\\"o\\":{\\"x\\":[0.333],\\"y\\":[0]},\\"n\\":[\\"0p667_1_0p333_0\\"],\\"t\\":-1,\\"s\\":[0],\\"e\\":[100]},{\\"t\\":37.0000015070409}],\\"ix\\":2},\\"o\\":{\\"a\\":0,\\"k\\":0,\\"ix\\":3},\\"m\\":1,\\"ix\\":2,\\"nm\\":\\"Trim Paths 1\\",\\"mn\\":\\"ADBE Vector Filter - Trim\\"}],\\"ip\\":0,\\"op\\":50.0000020365418,\\"st\\":-1.00000004073083,\\"bm\\":0,\\"sr\\":1},{\\"ddd\\":0,\\"ind\\":2,\\"ty\\":4,\\"nm\\":\\"flamingo ring 3\\",\\"parent\\":1,\\"ks\\":{\\"o\\":{\\"a\\":0,\\"k\\":100},\\"r\\":{\\"a\\":1,\\"k\\":[{\\"i\\":{\\"x\\":[0.785],\\"y\\":[1]},\\"o\\":{\\"x\\":[1],\\"y\\":[0]},\\"n\\":[\\"0p785_1_1_0\\"],\\"t\\":17,\\"s\\":[14.2],\\"e\\":[360]},{\\"t\\":50.0000020365418}]},\\"p\\":{\\"a\\":0,\\"k\\":[0,0,0]},\\"a\\":{\\"a\\":0,\\"k\\":[0,0,0]},\\"s\\":{\\"a\\":0,\\"k\\":[100,100,100]}},\\"ao\\":0,\\"shapes\\":[{\\"ty\\":\\"gr\\",\\"it\\":[{\\"d\\":1,\\"ty\\":\\"el\\",\\"s\\":{\\"a\\":0,\\"k\\":[54,54]},\\"p\\":{\\"a\\":0,\\"k\\":[0,0]},\\"nm\\":\\"Ellipse Path 1\\",\\"mn\\":\\"ADBE Vector Shape - Ellipse\\"},{\\"ty\\":\\"st\\",\\"c\\":{\\"a\\":0,\\"k\\":[0.435,0.812,0.592,1]},\\"o\\":{\\"a\\":0,\\"k\\":100},\\"w\\":{\\"a\\":0,\\"k\\":6},\\"lc\\":2,\\"lj\\":1,\\"ml\\":4,\\"nm\\":\\"Stroke 1\\",\\"mn\\":\\"ADBE Vector Graphic - Stroke\\"},{\\"ty\\":\\"tr\\",\\"p\\":{\\"a\\":0,\\"k\\":[0,0],\\"ix\\":2},\\"a\\":{\\"a\\":0,\\"k\\":[0,0],\\"ix\\":1},\\"s\\":{\\"a\\":0,\\"k\\":[100,100],\\"ix\\":3},\\"r\\":{\\"a\\":0,\\"k\\":0,\\"ix\\":6},\\"o\\":{\\"a\\":0,\\"k\\":100,\\"ix\\":7},\\"sk\\":{\\"a\\":0,\\"k\\":0,\\"ix\\":4},\\"sa\\":{\\"a\\":0,\\"k\\":0,\\"ix\\":5},\\"nm\\":\\"Transform\\"}],\\"nm\\":\\"Ellipse 1\\",\\"np\\":3,\\"mn\\":\\"ADBE Vector Group\\"},{\\"ty\\":\\"tm\\",\\"s\\":{\\"a\\":1,\\"k\\":[{\\"i\\":{\\"x\\":[0.833],\\"y\\":[0.833]},\\"o\\":{\\"x\\":[0.167],\\"y\\":[0.167]},\\"n\\":[\\"0p833_0p833_0p167_0p167\\"],\\"t\\":42,\\"s\\":[0],\\"e\\":[1]},{\\"t\\":44.0000017921567}],\\"ix\\":1},\\"e\\":{\\"a\\":1,\\"k\\":[{\\"i\\":{\\"x\\":[0.833],\\"y\\":[1]},\\"o\\":{\\"x\\":[0.333],\\"y\\":[0]},\\"n\\":[\\"0p833_1_0p333_0\\"],\\"t\\":20,\\"s\\":[0],\\"e\\":[1]},{\\"t\\":44.0000017921567}],\\"ix\\":2},\\"o\\":{\\"a\\":0,\\"k\\":0,\\"ix\\":3},\\"m\\":1,\\"ix\\":2,\\"nm\\":\\"Trim Paths 1\\",\\"mn\\":\\"ADBE Vector Filter - Trim\\"}],\\"ip\\":17.0000006924242,\\"op\\":44.0000017921567,\\"st\\":-1.00000004073083,\\"bm\\":0,\\"sr\\":1},{\\"ddd\\":0,\\"ind\\":3,\\"ty\\":4,\\"nm\\":\\"flamingo ring 2\\",\\"parent\\":1,\\"ks\\":{\\"o\\":{\\"a\\":0,\\"k\\":100},\\"r\\":{\\"a\\":1,\\"k\\":[{\\"i\\":{\\"x\\":[0.612],\\"y\\":[1]},\\"o\\":{\\"x\\":[1],\\"y\\":[0]},\\"n\\":[\\"0p612_1_1_0\\"],\\"t\\":17,\\"s\\":[14.2],\\"e\\":[360]},{\\"t\\":50.0000020365418}]},\\"p\\":{\\"a\\":0,\\"k\\":[0,0,0]},\\"a\\":{\\"a\\":0,\\"k\\":[0,0,0]},\\"s\\":{\\"a\\":0,\\"k\\":[100,100,100]}},\\"ao\\":0,\\"shapes\\":[{\\"ty\\":\\"gr\\",\\"it\\":[{\\"d\\":1,\\"ty\\":\\"el\\",\\"s\\":{\\"a\\":0,\\"k\\":[54,54]},\\"p\\":{\\"a\\":0,\\"k\\":[0,0]},\\"nm\\":\\"Ellipse Path 1\\",\\"mn\\":\\"ADBE Vector Shape - Ellipse\\"},{\\"ty\\":\\"st\\",\\"c\\":{\\"a\\":0,\\"k\\":[0.435,0.812,0.592,1]},\\"o\\":{\\"a\\":0,\\"k\\":100},\\"w\\":{\\"a\\":0,\\"k\\":6},\\"lc\\":2,\\"lj\\":1,\\"ml\\":4,\\"nm\\":\\"Stroke 1\\",\\"mn\\":\\"ADBE Vector Graphic - Stroke\\"},{\\"ty\\":\\"tr\\",\\"p\\":{\\"a\\":0,\\"k\\":[0,0],\\"ix\\":2},\\"a\\":{\\"a\\":0,\\"k\\":[0,0],\\"ix\\":1},\\"s\\":{\\"a\\":0,\\"k\\":[100,100],\\"ix\\":3},\\"r\\":{\\"a\\":0,\\"k\\":0,\\"ix\\":6},\\"o\\":{\\"a\\":0,\\"k\\":100,\\"ix\\":7},\\"sk\\":{\\"a\\":0,\\"k\\":0,\\"ix\\":4},\\"sa\\":{\\"a\\":0,\\"k\\":0,\\"ix\\":5},\\"nm\\":\\"Transform\\"}],\\"nm\\":\\"Ellipse 1\\",\\"np\\":3,\\"mn\\":\\"ADBE Vector Group\\"},{\\"ty\\":\\"tm\\",\\"s\\":{\\"a\\":1,\\"k\\":[{\\"i\\":{\\"x\\":[0.833],\\"y\\":[0.833]},\\"o\\":{\\"x\\":[0.167],\\"y\\":[0.167]},\\"n\\":[\\"0p833_0p833_0p167_0p167\\"],\\"t\\":42,\\"s\\":[0],\\"e\\":[13.7]},{\\"t\\":44.0000017921567}],\\"ix\\":1},\\"e\\":{\\"a\\":1,\\"k\\":[{\\"i\\":{\\"x\\":[0.833],\\"y\\":[1]},\\"o\\":{\\"x\\":[0.333],\\"y\\":[0]},\\"n\\":[\\"0p833_1_0p333_0\\"],\\"t\\":20,\\"s\\":[0],\\"e\\":[13.7]},{\\"t\\":44.0000017921567}],\\"ix\\":2},\\"o\\":{\\"a\\":0,\\"k\\":0,\\"ix\\":3},\\"m\\":1,\\"ix\\":2,\\"nm\\":\\"Trim Paths 1\\",\\"mn\\":\\"ADBE Vector Filter - Trim\\"}],\\"ip\\":17.0000006924242,\\"op\\":44.0000017921567,\\"st\\":-1.00000004073083,\\"bm\\":0,\\"sr\\":1},{\\"ddd\\":0,\\"ind\\":4,\\"ty\\":4,\\"nm\\":\\"flaming ring 1\\",\\"parent\\":1,\\"ks\\":{\\"o\\":{\\"a\\":0,\\"k\\":100},\\"r\\":{\\"a\\":0,\\"k\\":0},\\"p\\":{\\"a\\":0,\\"k\\":[0,0,0]},\\"a\\":{\\"a\\":0,\\"k\\":[0,0,0]},\\"s\\":{\\"a\\":0,\\"k\\":[100,100,100]}},\\"ao\\":0,\\"shapes\\":[{\\"ty\\":\\"gr\\",\\"it\\":[{\\"d\\":1,\\"ty\\":\\"el\\",\\"s\\":{\\"a\\":0,\\"k\\":[54,54]},\\"p\\":{\\"a\\":0,\\"k\\":[0,0]},\\"nm\\":\\"Ellipse Path 1\\",\\"mn\\":\\"ADBE Vector Shape - Ellipse\\"},{\\"ty\\":\\"st\\",\\"c\\":{\\"a\\":0,\\"k\\":[0.435,0.812,0.592,1]},\\"o\\":{\\"a\\":0,\\"k\\":100},\\"w\\":{\\"a\\":0,\\"k\\":6},\\"lc\\":2,\\"lj\\":1,\\"ml\\":4,\\"nm\\":\\"Stroke 1\\",\\"mn\\":\\"ADBE Vector Graphic - Stroke\\"},{\\"ty\\":\\"tr\\",\\"p\\":{\\"a\\":0,\\"k\\":[0,0],\\"ix\\":2},\\"a\\":{\\"a\\":0,\\"k\\":[0,0],\\"ix\\":1},\\"s\\":{\\"a\\":0,\\"k\\":[100,100],\\"ix\\":3},\\"r\\":{\\"a\\":0,\\"k\\":0,\\"ix\\":6},\\"o\\":{\\"a\\":0,\\"k\\":100,\\"ix\\":7},\\"sk\\":{\\"a\\":0,\\"k\\":0,\\"ix\\":4},\\"sa\\":{\\"a\\":0,\\"k\\":0,\\"ix\\":5},\\"nm\\":\\"Transform\\"}],\\"nm\\":\\"Ellipse 1\\",\\"np\\":3,\\"mn\\":\\"ADBE Vector Group\\"},{\\"ty\\":\\"tm\\",\\"s\\":{\\"a\\":1,\\"k\\":[{\\"i\\":{\\"x\\":[0.667],\\"y\\":[1]},\\"o\\":{\\"x\\":[1],\\"y\\":[0]},\\"n\\":[\\"0p667_1_1_0\\"],\\"t\\":8,\\"s\\":[0],\\"e\\":[100]},{\\"t\\":48.0000019550801}],\\"ix\\":1},\\"e\\":{\\"a\\":1,\\"k\\":[{\\"i\\":{\\"x\\":[0.667],\\"y\\":[1]},\\"o\\":{\\"x\\":[1],\\"y\\":[0]},\\"n\\":[\\"0p667_1_1_0\\"],\\"t\\":-1,\\"s\\":[0],\\"e\\":[100]},{\\"t\\":37.0000015070409}],\\"ix\\":2},\\"o\\":{\\"a\\":0,\\"k\\":0,\\"ix\\":3},\\"m\\":1,\\"ix\\":2,\\"nm\\":\\"Trim Paths 1\\",\\"mn\\":\\"ADBE Vector Filter - Trim\\"}],\\"ip\\":15.0000006109625,\\"op\\":44.0000017921567,\\"st\\":-1.00000004073083,\\"bm\\":0,\\"sr\\":1}]}"
        speed={1}
        style={
          Object {
            "alignSelf": "center",
            "aspectRatio": 1,
            "height": "100%",
            "width": 86.42857142857142,
          }
        }
        testID="loading"
        variant="button"
      />
    </View>
  </View>
</View>
`;
